File: StructuralPatterns/Decorator/CSharp/FormattingDecorator.cs
namespace Decorator {
    public class FormattingDecorator : NotifierDecorator {
        public FormattingDecorator(INotifier wrapper) : base(wrapper) {}
        public override void Send(string message) {
            var formatted = $"*** {message.ToUpper()} ***";
            base.Send(formatted);
        }
    }
}

File: StructuralPatterns/Decorator/Go/formatting_decorator.go
package main

import (
	"fmt"
	"strings"
)

type FormattingDecorator struct {
	NotifierDecorator
}

func NewFormattingDecorator(notifier Notifier) *FormattingDecorator {
	return &FormattingDecorator{NotifierDecorator{wrapped: notifier}}
}

func (d *FormattingDecorator) Send(message string) {
	formatted := fmt.Sprintf("*** %s ***", strings.ToUpper(message))
	d.wrapped.Send(formatted)
}
